cmd/neofs-node/policy.go

```diff
@@ -152,3 +153,45 @@ func (x *containerPolicyContext) applyAtEpoch(epoch uint64, cache *lru.Cache[con
 	cache.Add(cacheKey, result)
 	return result, nil
 }
+
+// getNodesForObject reads storage policy of the referenced container from the
+// underlying container storage, reads network map at the specified epoch from
+// the underlying storage, applies the storage policy to it and returns sorted
+// lists of selected storage nodes along with the per-list numbers of primary
+// object holders. Resulting slices must not be changed.
+func (x *containerNodes) getNodesForObject(addr oid.Address) ([][]netmapsdk.NodeInfo, []uint, error) {
+	epoch, err := x.network.Epoch()
+	if err != nil {
+		return nil, nil, fmt.Errorf("read current NeoFS epoch: %w", err)
+	}
+	cnrID := addr.Container()
+	cnr, err := x.containers.Get(cnrID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("read container by ID: %w", err)
+	}
+	networkMap, err := x.network.GetNetMapByEpoch(epoch)
+	if err != nil {
+		return nil, nil, fmt.Errorf("read network map at epoch #%d: %w", epoch, err)
+	}
+
+	policy := cnr.Value.PlacementPolicy()
+	nodeLists, err := networkMap.ContainerNodes(policy, cnrID)
+	if err != nil {
+		return nil, nil, fmt.Errorf("apply container's storage policy to the network map at epoch #%d: %w", epoch, err)
+	} else if nodeLists, err = networkMap.PlacementVectors(nodeLists, addr.Object()); err != nil {
```

Comment:
why `else if`? just a regular algorithm, step by step, stop on any error. additional nesting it not required

Reply:
idk what to say, just a code style

Reply:
we have already discussed it, leaving the same link is the only thing i can do here: https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

Reply:
this one is still arguable

Reply:
elif no more

Reply:
oh, i see. but no GH's `compare` says me about it sadly